sceptre/resolvers/stack_output.py

```diff
@@ -167,3 +167,42 @@ def resolve(self):
         return self._get_output_value(
             dependency_stack_name, output_key, profile
         )
+
+
+class StackOutputExternalRegion(StackOutputBase):
+    """
+    Resolver for retrieving the value of an output of any Stack in any region.
+
+    :param argument: The Stack name, output name, and region to get.
+    :type argument: str in the format ``"<full stack name>::<output key>"``
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(StackOutputExternalRegion, self).__init__(*args, **kwargs)
+
+    def resolve(self):
+        """
+        Retrieves the value of CloudFormation output of the external Stack
+
+        :returns: The value of the Stack output.
+        :rtype: str
+        """
+
+        self.logger.debug(
+            "Resolving external Stack output: {0}".format(self.argument)
+        )
+
+        profile = None
+        arguments = shlex.split(self.argument)
+
+        stack_argument = arguments[0]
+        region = arguments[1]
```

Comment:
I thought about this quite a bit. Those are positional arguments, so if I wanted to specify a region I would _have_ to also specify a profile name, thus that argument becomes non-optional. We could have some logic to figure out if the second argument is a region or not, either through a regex or whitelist. However that approach seems like it would take more effort to maintain. I also couldn't think of a good way to make them non-positional without breaking backwards compatibility.

Overall, I don't think adding a separate resolver is ideal, but I think it's the best option if we want to keep backwards compatibility for `StackOutputExternal`. Open to ideas though.